apps/web/src/app/_components/companies/all-company-roles.tsx

```diff
@@ -15,24 +15,31 @@ export default function RenderAllRoles({ company }: RenderAllRolesProps) {
   return (
     <>
       <h2 className="mb-4 text-2xl">Job Postings</h2>
-      {roles.isSuccess && roles.data.length > 0 ? (
+      {roles.isPending ? (
+        <LoadingResults />
+      ) : (
         <div className="mb-8 grid h-[30dvh] w-full grid-cols-1 gap-3 p-1 md:grid-cols-2 xl:grid-cols-3">
-          {roles.data.map((role) => {
-            return (
-              <div key={role.id} className="p-2">
-                <RoleCardPreview
-                  reviewObj={role}
-                  className={cn("mb-4 hover:bg-cooper-gray-100")}
-                />
-              </div>
-            );
-          })}
+          {roles.isSuccess && roles.data.length > 0 ? (
+            <>
+              {roles.data.map((role) => {
+                return (
+                  <div key={role.id} className="p-2">
+                    <RoleCardPreview
+                      reviewObj={role}
+                      className={cn("mb-4 hover:bg-cooper-gray-100")}
+                    />
+                  </div>
+                );
+              })}
+            </>
+          ) : null}
+          {company ? (
```

Comment:
Instead of a ternary operator with a null as the false value, I think we can use 

`company && ... `